nestkernel/target.h

```diff
@@ -165,73 +165,73 @@ inline Target::Target( const thread tid,
 inline void
 Target::set_lcid( const index lcid )
 {
-  assert( lcid < max_lcid_ );
+  assert( lcid < MAX_LCID );
   // Reset corresponding bits using complement of mask and write new
   // bits by shifting input appropriately. Need to cast to long first,
   // to avoid overflow of input by left shifts.
-  data_ = ( data_ & ( ~lcid_mask ) )
-    | ( static_cast< unsigned long >( lcid ) << lcid_shift );
+  data_ = ( data_ & ( ~MASK_LCID ) )
+    | ( static_cast< uint64_t >( lcid ) << SHIFT_LCID );
 }
 
 inline index
 Target::get_lcid() const
 {
-  return ( data_ & lcid_mask ) >> lcid_shift;
+  return ( data_ & MASK_LCID ) >> SHIFT_LCID;
 }
 
 inline void
 Target::set_rank( const thread rank )
 {
-  assert( rank < max_rank_ );
-  data_ = ( data_ & ( ~rank_mask ) )
-    | ( static_cast< unsigned long >( rank ) << rank_shift );
+  assert( rank < MAX_RANK );
+  data_ = ( data_ & ( ~MASK_RANK ) )
+    | ( static_cast< uint64_t >( rank ) << SHIFT_RANK );
 }
 
 inline thread
 Target::get_rank() const
 {
-  return ( data_ & rank_mask ) >> rank_shift;
+  return ( data_ & MASK_RANK ) >> SHIFT_RANK;
 }
 
 inline void
 Target::set_tid( const thread tid )
 {
-  assert( tid < max_tid_ );
-  data_ = ( data_ & ( ~tid_mask ) )
-    | ( static_cast< unsigned long >( tid ) << tid_shift );
+  assert( tid < MAX_TID );
+  data_ =
+    ( data_ & ( ~MASK_TID ) ) | ( static_cast< uint64_t >( tid ) << SHIFT_TID );
 }
 
 inline thread
```

Comment:
Just realized: anyone checked that the `data_` top bit and the masks following guarantee either that the top bit is 0 after masking, or that the return cast throws away any top 1s created by right shifting?

Reply:
I am not sure if I understand your comment correctly. I think the author's intention is to set the thread-id (the bits defined by the bit-mask) without touching anything else. The high-order "processed-bit" is toggled elsewhere and must not be changed here.

Reply:
Checked: remote_target_id_ is unsigned.